src/vos/vos_aggregate.c

```diff
@@ -1712,9 +1721,12 @@ enqueue_phy_ent(struct agg_merge_window *mw, struct evt_extent *phy_ext,
 		const vos_iter_entry_t *entry, bio_addr_t *addr,
 		struct dcs_csum_info *csum_info, uint32_t ver)
 {
-	struct agg_phy_ent *phy_ent;
+	struct agg_phy_ent	*phy_ent;
+	daos_size_t		 phy_ent_size = sizeof(*phy_ent);
 
-	D_ALLOC_PTR(phy_ent);
+	/* Will append the checksum to the end of the phy_ent structure */
+	phy_ent_size += csum_info->cs_buf_len;
```

Comment:
Oh, you're right, I did miss the csum copy, thank you. I'll fix that. I started using the on-disk csum info, but ran into an issue where valgrind detected that memory being freed before the checksum comparison (which accesses the memory). I didn't totally understand the issue (if I remember right, the free was happening in a umem_tx_commit), but by not using the on-disk csum info the problem was resolved.

Reply:
When you use the on-disk csum info, did you allocate csum buffer for truncated physical entry? In the old code, the csum of truncated physical entry will point to a freed memory region, but I think allocating & copy csum for truncated physical entry above (in insert_segments()) should fix the issue, right?

You know we were trying hard to reduce various unnecessary memory copies in aggregation (to minimize the performance impact), so it would be great if we can fix the defect without extra csum copies.

Reply:
Removed this memcpy and added the on-demand alloc you suggested.